user_external/lib/imap.php

```diff
@@ -49,6 +49,32 @@ public function checkPassword($uid, $password) {
 			imap_close($mbox);
 			$uid = mb_strtolower($uid);
 			$this->storeUser($uid);
+			
+			if (preg_match('/^([^@]+)@(.+)\.[^.]+$/', $uid, $matches)) { 
```

Comment:
>  not expert enough to change my patch

you mean with git? you can simply add a new commit and push the branch again.

I think you meant:
`if (\OC::$server->getMailer()->validateMailAddress($uid) && preg_match(...)) {` (&& operator)